WrapperValidation.gitlab-ci.yml

```diff
@@ -0,0 +1,7 @@
+validate-wrapper:
+  stage: .pre
+  image:
+    name: ghcr.io/aepfli/wrapper-validation:latest
```

Comment:
Shall we change this to `ghcr.io/gradle` if we use `ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME` in the publish job?

Reply:
yes we can, this is currently just for evaluation, that it works, and to show the greater picture - the question is, what kind of hosting for the docker image we will be using.
- ghcr
- docker hub
- gitlab registry (with an automatically updated fork of this project)

There are multiple possible solutions, and i am open to any of those ;)

Reply:
I don't think any of us have enough experience with GitLab to make any sort of evaluation as to the value of picking one vs the other. I think that a clear list of pros-and cons would be useful. I'd defer to your suggestion here most likely.

Reply:
# Comparison of Registries for the docker image

Options:
- Docker Hub
- GitHub Docker Registry
- GitLab Docker Registry

## Docker Hub

### Pros:

- easiest accessible registry - image name would be just `gradle/wrapper-validation`
- account does already exist
- easy do apply to all kind of docker based CI systems 

### Cons:

- secrets need to be added to GitHub for pushing*
- pricing - i have no clue about the impact regarding this

* as far as i understood, there is already a teamcity server in place, which pushes the other docker images, but gradle wrapper validation is using GitHub Actions, so we need to somehow provide this credentials.

## GitHub Docker Registry

### Pros:

- easily included in the GitHub Action
- Registry is already existing, as it is coupled to the repository
- as long as it is public the registry is free: 
	> GitHub Packages usage is free for public packages. For private packages, each account on GitHub.com receives a certain amount of free storage and data transfer, depending on the product used with the account. Any usage beyond the included amounts is controlled by spending limits. - https://docs.github.com/en/billing/managing-billing-for-github-packages/about-billing-for-github-packages

### Cons:

- image name for includes will not be as short as intended
- not as easy to find, people will search docker hub for such images


## GitLab Docker Registry

### Pros:

- Registry is easily included into GitLab Actions
- Due to the Fork (see cons) the include statement will be much clearer for gitlab ci
	```yaml
	include:
		- remote: https://raw.githubusercontent.com/gradle/wrapper-validation-action/master/WrapperValidation.gitlab-ci.yml
	```
	turns into
	```yaml
	include:
		- project: <Project Path eg. gradle/wrapper-validation - depends on availability of names>
		  file: WrapperValidation.gitlab-ci.yml
	```
- docker image creation could be also done via GitLab Registry
- as longas it is public, registry is free
- 

### Cons:

- Fork needs to be created on GitLab.com side (i can maintain and create it) - i would recommend a fork which is autmatically updating itself.
- Additional indirection with GitLab
- image name for includes will not be as short as intended
- not as easy to find, people will search docker hub for such images


## Recommendation

There are two scenarios, which i can recommend for now, which actually could also be applied in parallel

1. i would go with GitHub Registry - it is easy to apply and as a first step a good evaluation. We will see adaption etc.

2. i would also prepare a GitLab CI file to push it to a gitlab registry, I can create a group and a fork, which automatically updates from github (so source of truth will still be GitHub) - this would allow users to use the easier and better readable approach for includes.


I will prepare 1. during the christmas holidays, and i will also update the readme. - we can then decide how to proceed in the new year.

Reply:
I think I agree with your recommendation to go with the GitHub Registry. Thank you for writing up this thorough assessment, much appreciated.